src/overlays/actors/ovl_En_Dodongo/z_en_dodongo.c

```diff
@@ -136,97 +279,837 @@ static DamageTable D_808792D4 = {
+    if (func_8087721C(this)) {
+        func_800B9010(&this->actor, NA_SE_EN_DODO_J_FIRE - SFX_FLAG);
+        frame = this->skelAnime.animCurrentFrame - 29.0f;
+        end = frame >> 1;
+        if (end >= 4) {
+            end = 3;
+        }
+
+        element = &this->collider3.elements[0];
+        temp_f2 = Math_SinS(this->actor.shape.rot.y) * this->unk_334;
+        temp_f12 = Math_CosS(this->actor.shape.rot.y) * this->unk_334;
+
+        for (i = 0; i < end; i++, element++) {
+            element->dim.worldSphere.center.x = this->unk_348[0].x + (element->dim.modelSphere.center.z * temp_f2);
+            element->dim.worldSphere.center.y =
+                this->unk_348[0].y + (element->dim.modelSphere.center.y * this->unk_334);
+            element->dim.worldSphere.center.z = this->unk_348[0].z + (element->dim.modelSphere.center.z * temp_f12);
+            element->dim.worldSphere.radius = element->dim.modelSphere.radius;
+        }
+
+        D_80879354.y = this->unk_334 * -4.5f;
+        D_80879348.x = 2.5f * temp_f2;
+        D_80879348.y = this->unk_334 * 1.4f;
+        D_80879348.z = 2.5f * temp_f12;
+        EffectSsDFire_Spawn(globalCtx, &this->unk_348[0], &D_80879354, &D_80879348, this->unk_334 * 100.0f,
+                            this->unk_334 * 35.0f, 0xFF - (frame * 10), 5, 0, 8);
+    } else if ((this->skelAnime.animCurrentFrame >= 2.0f) && (this->skelAnime.animCurrentFrame <= 20.0f)) {
+        func_800B9010(&this->actor, NA_SE_EN_DODO_J_BREATH - SFX_FLAG);
+    }
+
+    if (SkelAnime_FrameUpdateMatrix(&this->skelAnime)) {
+        func_80877D50(this);
+    }
+}
+
+void func_80877D50(EnDodongo* this) {
+    SkelAnime_ChangeAnimDefaultStop(&this->skelAnime, &D_06003088);
+    this->actionFunc = func_80877D90;
+}
+
+void func_80877D90(EnDodongo* this, GlobalContext* globalCtx) {
+    if (SkelAnime_FrameUpdateMatrix(&this->skelAnime)) {
+        func_808773C4(this);
+        this->timer = Rand_S16Offset(10, 20);
+    }
+}
+
+void func_80877DE0(EnDodongo* this) {
+    SkelAnime_ChangeAnim(&this->skelAnime, &D_060028F0, -1.0f, 35.0f, 0.0f, 2, -4.0f);
+    this->actor.flags |= 0x10;
+    this->timer = 25;
+    this->actionFunc = func_80877E60;
+    this->actor.speedXZ = 0.0f;
+}
+
+void func_80877E60(EnDodongo* this, GlobalContext* globalCtx) {
+    static Vec3f D_80879360 = { 0.0f, 0.6f, 0.0f };
+    static Color_RGBA8 D_8087936C = { 255, 255, 255, 255 };
+    Vec3f sp84;
+    Vec3f sp78;
+    s16 i;
+    Actor* explosive = globalCtx->actorCtx.actorList[ACTORCAT_EXPLOSIVES].first;
+    Vec3f sp64;
+    s32 pad;
+    s16 sp5E;
+    s16 sp5C;
+
+    while (explosive != NULL) {
+        if (explosive == this->actor.child) {
+            break;
+        }
+        explosive = explosive->next;
+    }
+
+    if (explosive == NULL) {
+        this->actor.child = NULL;
+    }
+
+    if (this->actor.child != NULL) {
+        Math_Vec3f_Copy(&this->actor.child->world.pos, &this->unk_308);
+        this->actor.child->freezeTimer = 30000;
```

Comment:
Is this timer suppose to be (20 frames / s) * (60 s / min) * (25 min), so this is suppose to be 25 minutes? Is my math right? If so, could you leave a comment `// 25 min`?

Applies below for another freezeTimer too